client/src/api/block_builder/input_selection/core/requirement/alias.rs

```diff
@@ -118,16 +118,7 @@ impl InputSelection {
         // PANIC: safe to unwrap as it's been checked that both can't be None at the same time.
         let input = selected_input.unwrap_or_else(|| &self.available_inputs[available_index.unwrap()]);
 
-        if is_alias_state_transition(
-            input,
-            // TODO this output conversion is temporarily needed to accommodate the current version of ISA.
-            self.outputs
-                .iter()
-                .map(|output| output.output.clone())
-                .collect::<Vec<_>>()
-                .as_slice(),
-        )? == Some((false, true))
-        {
+        if is_alias_state_transition(input, &self.outputs)? == Some((false, true)) {
```

Comment:
This double bool is confusing, maybe we should change to a struct or enum
Do you want to add this as point to Input Selection Algorithm improvements or should I create an own issue?

Reply:
Yeah I don't like it either. I initially did it this way because the function already existed and I wanted it to be compatible with the previous ISA to not break everything while I work on the new one. I'll add it to my issue and refactor it later.

Reply:
Added

Reply:
Theres no struct or enum here yet :) Is that intended